### DISCUSSION TITLE
Deployment of Nephio on Physical Servers for CI Testing

### DISCUSSION THREAD

**Discussion by arora-sagar** [nephio-project/nephio#752] [Ideas]
[Anuket lab of UNH](https://laas.anuket.iol.unh.edu/) can provide Nephio resources to create the same [topology of R2](https://github.com/nephio-project/docs/blob/main/static/images/user-guides/nephio-r2-deployment-diagram.png) and [topology of R1](https://github.com/nephio-project/docs/blob/main/static/images/user-guides/nephio-r1-5g-network.png). 

Here is what I think we can start with:

- In the beginning, we can start by reserving four servers in Anuket Lab, one for the Nephio MGMT cluster, three for the workload cluster, and reserving 10 VLANs. 
- Pre-installed OS (Ubuntu or Fedora) on Nephio MGMT cluster
- Pre-installed K8s cluster (Any distribution of K8s) for Nephio mgmt cluster
- Option 1: Pre-installed OS on workload cluster and create a cluster using Nephio
- Option 2: OS installation and cluster creation via Nephio (We can take the help of Sylva)
- Networking will be manual unless we find an alternative to container lab 
- After every CI run we will have to un-install all the Nephio packages and the Nephio workload cluster 
- Test on physical servers can run weekly rather than triggering on every MR. 

The idea is to discuss and define the CI testbed.

Feel free to join the thread.

**Comment by henderiw** [comment]
I believe it is fine to have static VLANs assigned. We need to register these to the resource backend or use kuid.

**Comment by electrocucaracha** [comment]
What about to control the K8s installation of the workload clusters with [Kubean](https://kubean-io.github.io/kubean/en/)?
